query/src/main/scala/filodb/query/exec/rangefn/InstantFunction.scala

```diff
@@ -367,9 +367,16 @@ final case class HistogramBucketImpl() extends HistToDoubleIFunction {
   final def apply(value: Histogram, scalarParams: Seq[Double]): Double = {
     require(scalarParams.length == 1, "Bucket/le required for histogram bucket")
     val bucket = scalarParams(0)
-    for { b <- 0 until value.numBuckets optimized } {
-      if (Math.abs(value.bucketTop(b) - bucket) <= 1E-10) return value.bucketValue(b)
+    if (bucket == Double.PositiveInfinity) {
+      // Just get the top bucket if bucket scheme has +Inf at top, or return NaN
+      if (value.bucketTop(value.numBuckets - 1) == Double.PositiveInfinity) value.topBucketValue else Double.NaN
```

Comment:
else throw Invalid data exception ? This condition should never happen right ? Lets not silently return NaN and suppress the error - it wont surface. User will never know about it. It'll be harder to debug quantile problems

Reply:
Sure